src/runtime/threadpool/builder.rs

```diff
@@ -172,12 +186,19 @@ impl Builder {
             .enable_all()
             .on_thread_start(move || {
                 // We need the threadpool's sender to set up the default tokio
-                // 0.1 executor.
-                let lock = compat_sender2.read().unwrap();
+                // 0.1 executor. We also need to upgrade the weak pointer, if the
+                // pointer is no longer valid then the runtime has shutdown and the
+                // handle is no longer available.
+                //
+                // This upgrade will only fail if the compat runtime has been dropped.
+                let sender = compat_sender2.upgrade().expect("Compat runtime shutdown.");
```

Comment:
the upgrade failing is a bug, rather than a user error, correct? i'd prefer it if we could make it a little clearer in the panic message that this was an internal error. not a big deal as AFAICT this shouldn't ever actually happen?

Reply:
Yeah, agreed, let me fix this up.